src/metrics-gatherer.ts

```diff
@@ -135,45 +157,55 @@ export class MetricsGatherer {
 
 	// check that a metric is of the given type(s)
 	public checkMetricType(name: string, kinds: string[]) {
-		return kinds.includes(this.kinds[name]);
+		return kinds.includes(this.meta[name].kind);
+	}
+
+	public exists(name: string) {
+		// ensure meta exists, and use meta to determine kind, ensuring
+		// prometheus.Metric object of that kind exists
+		return this.meta[name] && this.metrics[this.meta[name].kind][name];
 	}
 
 	// used declaratively to ensure a given metric of a certain kind exists,
 	// given some custom params to instantiate it if absent
-	public ensureExists(name: string, kind: string, custom: CustomParams = {}) {
+	public ensureExists(
```

Comment:
Is there any reason for this to be public? I would expect this to be an internal function and the external ones are the `describe` and `counter`/etc functions, if it is internal only then it'd be nice to make it private as part of the major bump and then my PR can be a patch bump instead of a major

Reply:
Correct, it can be private. Will include